src/java.base/share/classes/java/lang/invoke/X-VarHandleSegmentView.java.template

```diff
@@ -25,546 +25,731 @@
+                (noStride ? Unaligned.FORM : Unaligned.OFFSET_FORM) :
+                (noStride ? Aligned.FORM : Aligned.OFFSET_FORM);
+#end[byte]
     }
 
-    @Override
-    final MethodType accessModeTypeUncached(VarHandle.AccessType accessType) {
-        return accessType.accessModeType(MemorySegment.class, $type$.class, MemoryLayout.class, long.class, long.class);
-    }
+    static {#if[byte]?final:sealed} class Unaligned {
 
-    @Override
-    public VarHandleSegmentAs$Type$s withInvokeExactBehavior() {
-        return hasInvokeExactBehavior() ?
-                this :
-                new VarHandleSegmentAs$Type$s(be, alignmentMask, true);
-    }
+        static final VarForm FORM = new VarForm(Unaligned.class, MemorySegment.class, $type$.class, long.class);
+        static final VarForm OFFSET_FORM = new VarForm(Unaligned.class, MemorySegment.class, $type$.class, long.class, long.class);
 
-    @Override
-    public VarHandleSegmentAs$Type$s withInvokeBehavior() {
-        return !hasInvokeExactBehavior() ?
-                this :
-                new VarHandleSegmentAs$Type$s(be, alignmentMask, false);
-    }
+        Unaligned() { throw new AssertionError(); }
 
-#if[floatingPoint]
-    @ForceInline
-    static $rawType$ convEndian(boolean big, $type$ v) {
-        $rawType$ rv = $Type$.$type$ToRaw$RawType$Bits(v);
-        return big == BE ? rv : $RawBoxType$.reverseBytes(rv);
-    }
+        @ForceInline
+        static $type$ get(VarHandle ob, Object obb, long base) {
+            return get(ob, obb, base, ((SegmentVarHandle) ob).offset);
+        }
 
-    @ForceInline
-    static $type$ convEndian(boolean big, $rawType$ rv) {
-        rv = big == BE ? rv : $RawBoxType$.reverseBytes(rv);
-        return $Type$.$rawType$BitsTo$Type$(rv);
-    }
+        @ForceInline
+        static $type$ get(VarHandle ob, Object obb, long base, long offset) {
+            SegmentVarHandle handle = (SegmentVarHandle)ob;
+            AbstractMemorySegmentImpl bb = handle.checkSegment(obb, base, true);
+#if[floatingPoint]
+            $rawType$ rawValue = SCOPED_MEMORY_ACCESS.get$RawType$Unaligned(bb.sessionImpl(),
+                    bb.unsafeGetBase(),
+                    offset(bb, base, offset),
+                    handle.be);
+            return $Type$.$rawType$BitsTo$Type$(rawValue);
 #else[floatingPoint]
 #if[byte]
-    @ForceInline
-    static $type$ convEndian(boolean big, $type$ n) {
-        return n;
-    }
+            return SCOPED_MEMORY_ACCESS.get$Type$(bb.sessionImpl(),
+                    bb.unsafeGetBase(),
+                    offset(bb, base, offset));
 #else[byte]
-    @ForceInline
-    static $type$ convEndian(boolean big, $type$ n) {
-        return big == BE ? n : $BoxType$.reverseBytes(n);
-    }
+            return SCOPED_MEMORY_ACCESS.get$Type$Unaligned(bb.sessionImpl(),
+                    bb.unsafeGetBase(),
+                    offset(bb, base, offset),
+                    handle.be);
```

Comment:
Why do we not have a call to `convEndian` here?

Reply:
This is just how it was. Refer to line 141 in old diff.

Reply:
Ah, right, the unaligned Unsafe routines support specifying the endianess already, but for the aligned variants it has to be emulated.